src/librustc_mir/interpret/operand.rs

```diff
@@ -345,7 +351,10 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M>
         mplace: MPlaceTy<'tcx, M::PointerTag>,
     ) -> EvalResult<'tcx, &str> {
         let len = mplace.len(self)?;
-        let bytes = self.memory.read_bytes(mplace.ptr, Size::from_bytes(len as u64))?;
+        let ptr = mplace.ptr.to_ptr()?;
```

Comment:
If the `len` is 0, this will ICE, whereas that case was previously working fine (`Memory::read_bytes` would handle length 0).

Reply:
It seems we do not have a test case for this, can you add one? Casting `1` into a `&[u8; 0]` and from there to `&str` should do it.

Reply:
`const FOO: &str = unsafe { &*(1_usize as *const [u8; 0] as *const [u8] as *const [str]) };` does not ICE

Reply:
Oh I just fail at testing. Now I get

```
error[E0080]: it is undefined behavior to use this value
  --> $DIR/int_ptr_for_zst_slices.rs:3:1
   |
LL | const FOO: &str = unsafe { &*(1_usize as *const [u8; 0] as *const [u8] as *const str) };
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered uninitialized or non-UTF-8 data in str at .<deref>
   |
   = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rust compiler repository if you believe it should not be considered undefined behavior

error: aborting due to previous error

For more information about this error, try `rustc --explain E0080`.
```